File: ants/bots/python/RandomBot.py
#!/usr/bin/env python
from random import shuffle
from ants import *

class RandomBot:
    def do_turn(self, ants):
        destinations = []
        for a_row, a_col in ants.my_ants():
            # try all directions randomly until one is passable and not occupied
            directions = AIM.keys()
            shuffle(directions)
            for direction in directions:
                (n_row, n_col) = ants.destination(a_row, a_col, direction)
                if (not (n_row, n_col) in destinations and
                        ants.passable(n_row, n_col)):
                    ants.issue_order((a_row, a_col, direction))
                    destinations.append((n_row, n_col))
                    break
            else:
                destinations.append((a_row, a_col))

if __name__ == '__main__':
    try:
        import psyco
        psyco.full()
    except ImportError:
        pass
    try:
        Ants.run(RandomBot())
    except KeyboardInterrupt:
        print('ctrl-c, leaving ...')

File: ants/bots/java/RandomBot.java
import java.util.*;

public class RandomBot implements Runnable {
	private static Ants ants = new Ants();
	public static void main(String[] args) {
		Ants.run(new RandomBot(), ants);
	}
	
	public void run() {
		Set<Tile> destinations = new HashSet<Tile>();
		for (Tile location : ants.myAnts()) {
			List<Aim> directions = new ArrayList<Aim>(EnumSet.allOf(Aim.class));
			Collections.shuffle(directions);
			boolean issued = false;
			for (Aim direction : directions) {
				Tile destination = ants.tile(location, direction);
				if (ants.ilk(destination).isUnoccupied() && !destinations.contains(destination)) {
					ants.issueOrder(location, direction);
					destinations.add(destination);
					issued = true;
					break;
				}
			}
			if (!issued) {
				destinations.add(location);
			}
		}
	}
}

